src/shims/time.rs

```diff
@@ -85,6 +86,33 @@ pub trait EvalContextExt<'mir, 'tcx: 'mir>: crate::MiriEvalContextExt<'mir, 'tcx
         Ok(0)
     }
 
+    #[allow(non_snake_case)]
+    fn GetSystemTimeAsFileTime(&mut self, LPFILETIME_op: OpTy<'tcx, Tag>) -> InterpResult<'tcx> {
+        let this = self.eval_context_mut();
+
+        this.assert_target_os("windows", "GetSystemTimeAsFileTime");
+        this.check_no_isolation("GetSystemTimeAsFileTime")?;
+
+        // A Windows tick is 100 nanoseconds. Windows epoch 1601-01-01T00:00:00Z
+        // is 11644473600 seconds before Unix epoch 1970-01-01T00:00:00Z.
+        let duration = system_time_to_duration(&SystemTime::now())?
+            .checked_add(Duration::from_secs(11644473600))
+            .unwrap();
+        let windows_nanoseconds_per_tick = 100;
+        let duration_ticks = u64::try_from(duration.as_nanos() / windows_nanoseconds_per_tick)
+            .map_err(|_| err_unsup_format!("programs running longer than 2^64 nanoseconds are not supported"))?;
+
+        let DWORD_tylayout = this.windows_ty_layout("DWORD")?;
```

Comment:
We are encoding DWORD = u32 everywhwere, so you might as well also do that here.
```suggestion
        let DWORD_tylayout = this.layout_of(this.tcx.types.u32)?;
```